x-pack/plugins/cases/public/components/create/flyout/use_cases_add_to_new_case_flyout.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CasesContextStoreActionsList } from '../../cases_context/cases_context_reducer';
+import { useCasesContext } from '../../cases_context/use_cases_context';
+import { CreateCaseFlyoutProps } from './create_case_flyout';
+
+export const useCasesAddToNewCaseFlyout = (props: CreateCaseFlyoutProps) => {
+  const context = useCasesContext();
+  const closeFlyout = () => {
```

Comment:
Do we need a `useCallback`?

Reply:
No.

Reply:
Generally speaking you want to memoize things that will cause unnecessary re-renders. In a perfect scenario nothing should be re-render if there's nothing to re-render.

That being said, memoizing has a cost: objects need to be checked and compared before deciding if they are cached or not and there's the comparison between what does it cost more, comparing that or re-rendering?

Generally speaking unless you have benchmarked and checked that those re-renders are expensive you should not try to over-optimize it. 

Simply re-rendering a react component should not be a reason to memoize things, there's need to know if that re-render is really affecting performance in some way. Re-renders are not as evil as all developers see them.

Reply:
Hi @academo , I agree with you about over-optimization is not good, sometimes it is overkill, but in this case, I am not convinced. Because any change of the context value will result in a new `closeFlyout` and `openFlyout` functions, which may cause unnecessary re-renders in the components that use them. 

These two functions only use the `dispatch` from the context, so there's no need to be re-created when another part of the context has changed (let's say `userCanCrud` for instance). This could be prevented by using `useCallback` with the `dispatch` function as a dependency.

And we know that the `dispatch` function will never change in the context value (it is set in the initialization and never updated), so these functions will also never change, preventing any unnecessary re-render in the components from any context change. Maybe this is a simple optimization that may be worth doing.

Reply:
I believe this is one of those optimizations that are totally unnecessary. I can do it if you insist of course

Reply:
> These hooks and components using them are generally used at the bottom of the render tree, where a re-render is a minimal performance impact: a popover, a single div with a button. Even if they generate re-render trying to over-optimize will create more CPU and Memory problems that what you are solving with this.

I also agree with that, but this is an assumption about how the other part will use it, we don't know that, so I think we should not do that assumption. Also we have also to take into account the cpu, the memory and garbage created with the re-creation of the functions.

I would be inclined to use the useCallback in this case, but I will not bother about it, I can see the points of doing the optimization and not doing it also. 👍

Reply:
Js is absolutely optimized for inline functions. To the point that recreating them is almost for free. I will add the useCallback but I see many cases like this and others where memoization is used unnecessary